src/packagers/JSPackager.js

```diff
@@ -14,6 +14,20 @@ const prelude = {
     .replace(/;$/, '')
 };
 
+const dedupKey = asset => {
+  // cannot rely only on generated JS for deduplication as paths like `../` can
+  // cause 2 identical JS files to be different depending on filesystem locations
+  return {
```

Comment:
We cannot use objects as keys for the dedupe map - they will never match since JS compares keys by object identity. You'll need to generate a string key.